lib/virt_autotest/utils.pm

```diff
@@ -68,8 +73,17 @@ sub restart_modular_libvirt_daemons {
         record_soft_failure("Restaring modular libvirt daemons has not been implemented in ALP. See poo#129086");
     }
     else {
-        systemctl("restart $_\{,-ro,-admin\}.socket") foreach @daemons;
-        systemctl("restart $_.service") foreach @daemons;
+        if (is_xen_host) {
```

Comment:
I would recommend a logic like this
```
if restart_modular_libvirt_daemons is called with empty parameter, set the default daemon list in above line 71, which differs for kvm and xen
elif restart_modular_libvirt_daemons is called with parameter, let @daemons be it, 

skip the @daemons modification here.
```

Reply:
Hmm, Alice 
look at the below logic: Are you recommending something like this?

`sub restart_modular_libvirt_daemons {
    my @daemons;

    if (@_ == 0) {
        if (is_xen_host) {
            @daemons = qw(virtxend virtstoraged virtnetworkd virtnodedevd virtsecretd virtproxyd virtnwfilterd);
        } else {
            @daemons = qw(virtqemud virtstoraged virtnetworkd virtnodedevd virtsecretd virtproxyd virtnwfilterd);
        }
    } else {
        @daemons = @_;
    }

    if (is_alp) {
        record_soft_failure("Restarting modular libvirt daemons has not been implemented in ALP. See poo#129086");
    }
    else {
        systemctl("restart $_\{,-ro,-admin\}.socket") foreach @daemons;
        systemctl("restart $_.service") foreach @daemons;
    }

    record_info("Libvirt daemons restarted", join(' ', @daemons));
}
`


here if the restart_modular_libvirt_daemons subroutine is called without any parameters, it checks if the host is a Xen host using the is_xen_host function. Based on the result, it sets the appropriate default daemon list (virtqemud for non-Xen hosts and virtxend for Xen hosts). If the subroutine is called with parameters, the @daemons array is assigned those parameters directly.

Reply:
Right :)

Reply:
thanks, pushed new commit

Reply:
@varunkojha Do you push successfully? I did not see the new code for this here.

Reply:
Yes Alice, I reverted back to test the difference between the logics.
Both versions achieve the same result, which is to determine the daemon list based on the input parameters or the host type. However, the earlier version (which we see now) using the ternary operator provides a more compact and readable implementation. It avoids the need for separate conditional statements and simplifies the code flow.

I agree that tests running into problems with the new changes has a low chance to be relevant in the different logical way of implementing the functions, I am checking on your suggestions.

Reply:
@alice-suse @Julie-CAO 

We have a successful run, and the suggestions worked.
[GuestInstallationKVM](https://openqa.opensuse.org/tests/3330711#)

Reply:
Pushed commit and tested

I first restart all the sockets for the daemons, allowing them some time to initialize properly. Introduced idle time between socket restarts and service restarts. This ensures that the sockets are fully operational before the services are restarted.

After the idle time, we proceed to restart the services for the daemons. This sequence helps avoid any potential issues related to the timing and initialization of the sockets and services and allows some time for the sockets to initialize properly before proceeding to restart the services.